pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,203 @@
+		logrus.Debugf("Trust bundle configmap reconcile request.Namespace/request.Name: '%s/%s'", request.Namespace, request.Name)
+		if err := r.client.Get(context.TODO(), request.NamespacedName, trustBundle); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("configmap not found; reconciliation will be skipped - request %v", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - just return without requeuing.
+			return reconcile.Result{}, nil
+		}
+
+		// Only proceed if request matches the configmap referenced by proxy trustedCA.
+		if err := r.configMapIsProxyTrustedCA(trustBundle.Name); err != nil {
+			logrus.Errorf("configmap '%s/%s' name differs from trustedCA of proxy '%s' or trustedCA not set; "+
+				"reconciliation will be skipped", trustBundle.Namespace, trustBundle.Name, proxyName)
+			return reconcile.Result{}, nil
+		}
+	} else {
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch the ClusterLogging instance
+	instance := &loggingv1.ClusterLogging{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-logging", Name: singletonName}, instance); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - just return without requeuing.
+		return reconcile.Result{}, nil
+	}
+
+	if instance.Spec.ManagementState == loggingv1.ManagementStateUnmanaged {
+		return reconcile.Result{}, nil
+	}
+
+	forwardinginstance := &logforwarding.LogForwarding{}
+	fiName := types.NamespacedName{Name: singletonName, Namespace: openshiftNS}
+	err := r.client.Get(context.TODO(), fiName, forwardinginstance)
+	if err != nil && !apierrors.IsNotFound(err) {
+		// Request object not found, could have been deleted after reconcile request.
+		// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	if err := k8shandler.ReconcileForGlobalProxy(instance, forwardinginstance, r.client); err != nil {
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// configMapIsProxyTrustedCA returns an error if cfgMapName does not match the
+// ConfigMap name referenced by proxy "cluster" trustedCA.
+func (r *ReconcileProxyConfig) configMapIsProxyTrustedCA(cfgMapName string) error {
+	proxyConfig := &configv1.Proxy{}
+	err := r.client.Get(context.TODO(), proxyNamespacedName(), proxyConfig)
+	if err != nil {
+		return fmt.Errorf("failed to get proxy '%s': %v", proxyName, err)
+	}
+
+	if proxyConfig.Spec.TrustedCA.Name != cfgMapName {
```

Comment:
why are you looking at proxyconfig.spec.trustedca.name?  you shouldn't care about that field or the configmap it points to.